core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -378,45 +368,24 @@ class JDBCSourceAsColumnarStore(override val connProperties: ConnectionPropertie
   }
 
   override def getConnection(id: String, onExecutor: Boolean): Connection = {
+    connectionType match {
+      case ConnectionType.Embedded =>
+        val currentCM = ContextService.getFactory.getCurrentContextManager
+        // val currentCM = ContextService.getFactory.newContextManager()
+        if (currentCM ne null) {
+          val conn = EmbedConnectionContext.getEmbedConnection(currentCM)
+          if (conn ne null) return conn
+        }
+      case _ =>
+    }
+
```

Comment:
@sumwale @vivekwiz 
If connectionType is Embedded and we are still returning Pooled connection then that needs to be closed in close Connection otherwise the pool connection will not be closed. We need to somehow indicate this to close connection method.